write-fonts/src/tables/instance_record.rs

```diff
@@ -0,0 +1,46 @@
+//! An fvar InstanceRecord
+
+#[allow(unused_imports)]
+use crate::codegen_prelude::*;
+
+// spec: If the value is 0xFFFF, then the value is ignored
+const PS_NAME_IGNORED: NameId = NameId::new(0xFFFF);
+
+/// The [InstanceRecord](https://learn.microsoft.com/en-us/typography/opentype/spec/fvar#instancerecord)
+#[derive(Debug, Clone, Default, PartialEq, Eq)]
+pub struct InstanceRecord {
+    /// The name ID for entries in the 'name' table that provide subfamily names for this instance.
+    pub subfamily_name_id: NameId,
+    /// Reserved for future use — set to 0.
+    pub flags: u16,
+    /// The coordinates array for this instance.
+    pub coordinates: Vec<Fixed>,
+    /// Optional. The name ID for entries in the 'name' table that provide PostScript names for this instance.
+    pub post_script_name_id: Option<NameId>,
+}
+
+impl FontWrite for InstanceRecord {
+    fn write_into(&self, writer: &mut TableWriter) {
+        self.subfamily_name_id.write_into(writer);
+        self.flags.write_into(writer);
+        self.coordinates.write_into(writer);
+        self.post_script_name_id
```

Comment:
This is generating the full instance record unconditionally even though the instance size is computed based on the presence of at least one non-None PostScript name. I think we’ll generate an invalid table if all are None.

Reply:
Note to self: trust None means do not emit and instead update this to be all Some / all none in set_instancesize.

Reply:
Fixed to only write if present. Having trouble getting validation wired; filed #308